include/linuxdeploy/core/desktopfileentry.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+// system headers
+#include <memory>
+#include <string>
+
+// library headers
+#include <boost/lexical_cast.hpp>
+
+namespace linuxdeploy {
+    namespace core {
+        namespace desktopfile {
+            class DesktopFileEntry {
```

Comment:
The desktop file class used to be a member as well. Since it returns such entries now, it must be in the namespace. Why shouldn't it be in there? If you can't provide a real reason, I'd leave them there. The whole thing will be extracted soon anyway, and there they will definitely be in the namespace.